lib/src/log/log_data.dart

```diff
@@ -0,0 +1,82 @@
+// Copyright 2016 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'log_fields.dart';
+
+/// LogData is data associated with a `Span`. Every LogData instance should
+/// specify at least one of Event and/or Payload.
+class LogData {
+  ///Returns a LogData instance.
+  LogData(
+    String event, {
+    DateTime timestamp,
+    String errorKind,
+    dynamic errorObject,
+    String message,
```

Comment:
I can be persuaded otherwise but I think this parameter should be required and positional.